docs/spl-docs/evaluation-functions/mathematical-functions.md

```diff
@@ -0,0 +1,441 @@
+
+The `<precision>` is optional, and if omitted the round function rounds to an integer.
+You cannot specify a negative number for the precision.
+
+### Example
+
+#### Specifying a value without precision
+
+The following example returns `n=4`. Because a `<precision>` is not specified, the number is rounded to the integer.
+
+```
+... | eval n=round(3.5)
+```
+
+#### Specifying a value and a precision
+
+The following example returns `n=2.56`.
+
+```
+... | eval n=round(2.55556, 2)
+```
+
+### Use-Case Example
+
+**Calculate average transaction amounts**
+
+**Problem:** A user wants to display the average transaction amounts in a financial report. The average amounts should be rounded to two decimal places for clarity and consistency in the report.
+
+**Solution:** The `round()` command can be used to round the average transaction amounts to two decimal places.
+
+```
+index=financial_transactions
+| stats avg(transaction_amount) as avg_amount by account_id
+| eval avg_amount_rounded = round(avg_amount, 2)
+| fields account_id, avg_amount, avg_amount_rounded
+```
+
+**Explanation:**
+1. The `stats` command calculates the average transaction amount for each account.
+2. The `eval` command rounds the average amount to two decimal places using `round(avg_amount, 2)`.
+
+This ensures that the average transaction amounts are displayed with two decimal places, making the financial report clear and consistent.
+
+## ln(\<num>)
+
+This function returns the natural logarithm of a number.
+
+### Usage
+
+The `<num>` argument can be the name of a numeric field or a numeric literal. 
+
+### Example
+
+The following example returns the natural logarithm of the values in the `bytes` field.
+
+```
+... | eval nat_logarithm=ln(bytes)
+```
+
+### Use-Case Example
+
+**Analyze exponential growth in website traffic**
+
+**Problem:** A user wants to analyze the exponential growth of website traffic over time. The natural logarithm can be used to transform the data, making it easier to identify trends and growth patterns.
+
+**Solution:** The `ln()` command can be used to calculate the natural logarithm of the number of website visits, which helps in analyzing growth trends.
+
+```
+index=web_traffic
+| stats sum(visits) as total_visits by date
+| eval log_visits = ln(total_visits)
+| fields date, total_visits, log_visits
+```
+
+**Explanation:**
+1. The `stats` command calculates the total number of website visits per day.
+2. The `eval` command calculates the natural logarithm of the total visits using `ln(total_visits)`.
+
+This allows the user to analyze the log-transformed data to identify exponential growth patterns in website traffic.
+
+## log(\<num>, \<base>)
+
+This function returns the logarithm of a number using a base. 
+
+### Usage
+
+The `<num>` argument can be the name of a numeric field or a numeric literal.
+
+The `<base>` is optional, and if omitted the log function uses base 10.
+
```

Comment:
There should be some restrictions on `num` and `base`. Mention those. Also mention what happens if those criteria are not met (does `log` give NULL?)

Reply:
So I have mentioned the restrictions. 
But regarding the error as of now we are throwing error on the backend, but we are not showing that on the UI.